document-portal/meson.build

```diff
@@ -0,0 +1,102 @@
+built_src_permission_store = gnome.gdbus_codegen('permission-store-dbus',
+    sources: '../data/org.freedesktop.impl.portal.PermissionStore.xml',
+    interface_prefix: 'org.freedesktop.impl.portal',
+    namespace: 'Xdg',
+    autocleanup: 'none',
+)
+
+src_db = files(
+    'permission-db.c',
+    'permission-db.h',
+    'gvdb/gvdb-reader.h',
+    'gvdb/gvdb-format.h',
+    'gvdb/gvdb-reader.c',
+    'gvdb/gvdb-builder.h',
+    'gvdb/gvdb-builder.c',
+)
+
+src_xdg_permission_store = [
+    src_xdp_utils,
+    src_db,
+    src_sd_escape,
+    built_src_permission_store,
+] + files(
+    'permission-store.c',
+    'xdg-permission-store.c',
+    'xdg-permission-store.h',
+)
+
+xdg_permission_store = executable('xdg-permission-store',
+    src_xdg_permission_store,
+    dependencies: deps_xdg_desktop_portal,
+    include_directories: incs_xdg_desktop_portal,
+    install: true,
+    install_dir: dir_libexecdir
+)
+
+configure_file(input: 'xdg-permission-store.service.in',
+    output: '@BASENAME@',
+    configuration: base_config,
+    install: true,
+    install_dir: dir_systemd_userunit,
+)
+
+built_src_document_portal = gnome.gdbus_codegen('document-portal-dbus',
+    sources: [
+        '../data/org.freedesktop.portal.Documents.xml',
```

Comment:
Probably want to use files() here.

Reply:
files() only has semantic meaning when assigned to a variable and reused in another subdir's meson.build.